external/fv3fit/fv3fit/data/tfdataset.py

```diff
@@ -27,16 +27,22 @@ def __post_init__(self):
             raise TypeError("times must be one of 'window' or 'start'")
 
     def get_record(self, name: str, ds: xr.Dataset, unstacked_dims: Sequence[str]):
+        for dim in unstacked_dims[:-1]:
+            if dim not in ds[name].dims:
+                raise ValueError("variable {} has no dimension {}".format(name, dim))
         if self.times == "start":
             ds = ds.isel(time=0)
-        data = stack(ds[name], unstacked_dims=unstacked_dims).values
+        dims = [d for d in unstacked_dims if d in ds[name].dims]
+        data = ds[name].transpose(*dims).values
         return data
 
 
 def open_zarr_using_filecache(url: str):
     cachedir = tempfile.mkdtemp()
     return xr.open_zarr(
-        "filecache::" + url, storage_options={"filecache": {"cache_storage": cachedir}}
+        "filecache::" + url,
+        storage_options={"filecache": {"cache_storage": cachedir}},
+        decode_times=False,
```

Comment:
Time can't be retrieved as a variable anyways, needed this to load 5-year long data.